nuxeo-csv-core/src/main/java/org/nuxeo/ecm/csv/core/CSVImporterImpl.java

```diff
@@ -19,24 +19,43 @@
 
 package org.nuxeo.ecm.csv.core;
 
-import org.nuxeo.ecm.csv.core.CSVImportLog.Status;
-import org.nuxeo.ecm.core.api.CoreSession;
-
 import java.io.File;
+import java.io.IOException;
 import java.util.Arrays;
 import java.util.List;
 import java.util.stream.Collectors;
 
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.ecm.core.api.Blobs;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.csv.core.CSVImportLog.Status;
+
 /**
  * @since 5.7
  */
 public class CSVImporterImpl implements CSVImporter {
 
+    private static final Log log = LogFactory.getLog(CSVImporterImpl.class);
+
     @Override
+    @Deprecated
     public String launchImport(CoreSession session, String parentPath, File csvFile, String csvFileName,
             CSVImporterOptions options) {
-        return new CSVImporterWork(session.getRepositoryName(), parentPath, session.getPrincipal().getName(), csvFile,
-                csvFileName, options).launch();
+        try {
+            return new CSVImporterWork(session.getRepositoryName(), parentPath, session.getPrincipal().getName(),
+                    Blobs.createBlob(csvFile), options).launch();
+        } catch (IOException e) {
```

Comment:
Sure. This refactoring is for https://jira.nuxeo.com/browse/NXP-21534. As you noticed I deprecated this method taking a File as argument. The point is we need to make sure the file content (associated blob) is stored in the transient store of the job before it goes async. Else this file might be deleted by GC before the jobs ends and ... boom.

The IOException is thrown by `Blobs.createBlob(csvFile)`. In the other `launchImport` I am directly taking the input blob of the operation so I don't have to care/deal with exception management. Ideally I wouldn't have caught this exception, but since I did not want to change the Work API for compat either, I came up with this catch/log.error statement. I agree that it is not the best thing to do but it is still better than what happened before (i.e. see description in https://jira.nuxeo.com/browse/NXP-21534)